File: src/Score/testParse.hs

import Text.Parsec
import Text.Parsec.String



p1 :: Parser String
p1 = 
  try p2
  <|>
  try p3
  <?> "p2 or p3"

p2 = string "foo"
p3 = string "food"


s = "foo"

main = putStrLn $ case parse p1 "" s of
  Left e -> show e
  Right v -> show v

File: app/Pyvid/testParse.py
from parse import Parser, parseFile
from graphemes import *


buf = '''\
[text id:3 x:100 y:100 font:"Times New Roman" size:16 bold:false italic:false
    show:true string:foo]
'''

def main():
   b = Parser(buf)
   lines = parseFile(b)
   for keyword, fields in lines:
      if keyword == "text":
         t = Text.read(fields)
         print( t.show().toString())
      else:
         print("Unknown keyword '%s'" % keyword)
      

main()

